sw/host/opentitanlib/src/util/file.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use nix::libc::c_int;
+use nix::poll;
+use std::convert::TryInto;
+use std::io;
+use std::os::unix::io::AsRawFd;
+use std::time::Duration;
+
+/// Waits for an event on `fd` or for `timeout` to expire.
+pub fn wait_timeout(fd: &impl AsRawFd, events: poll::PollFlags, timeout: Duration) -> Result<()> {
+    let timeout = timeout.as_millis().try_into().unwrap_or(c_int::MAX);
+    let mut pfd = [poll::PollFd::new(fd.as_raw_fd(), events)];
+    match poll::poll(&mut pfd, timeout) {
```

Comment:
Maybe add a `?` operator here?

Reply:
https://github.com/lowRISC/opentitan/pull/6859#discussion_r647458995
=)

The `?` operator would make the `Err` case in the match implicit, yes?

Reply:
Yup.

Reply:
Done.